lint-tasks.yml

```
@@ -128,6 +153,611 @@ tasks:
     internal: true
     cmd: "{{.ROOT_DIR}}/tools/yscope-dev-utils/lint-configs/symlink-cpp-lint-configs.sh"
 
+  check-cpp-static-diff:
+    sources: *cpp_source_files
+    deps:
+      - "cpp-lint-configs"
+      - ":core-generate"
+      - "venv"
+    cmds:
+      - task: ":utils:cpp-lint:clang-tidy-diff"
+        vars:
+          FLAGS:
+            - "--config-file '{{.ROOT_DIR}}/.clang-tidy'"
+            - "-p '{{.G_CORE_COMPONENT_BUILD_DIR}}'"
+          OUTPUT_DIR: "{{.G_LINT_CLANG_TIDY_DIR}}"
+          VENV_DIR: "{{.G_LINT_VENV_DIR}}"
+
+  check-cpp-static-full:
+    sources: *cpp_source_files
+    deps:
+      - "cpp-lint-configs"
+      - ":core-generate"
+      - "venv"
+    cmds:
+      - task: ":utils:cpp-lint:clang-tidy-find"
+        vars:
+          # NOTE: We do not lint `.inc` files because they result in false positives due to circular
+          # includes and we plan to remove them going forward.
+          #
+          # The only error in the following files is a circular include due to including
+          # decoding_methods.hpp (decoding_methods.hpp and decoding_methods.inc include each other)
+          # and should be removed from the exclude list once decoding_methods.inc is removed:
+          #
+          # - src/clp/ffi/ir_stream/ir_unit_deserialization_methods.hpp
+          # - src/clp/ffi/ir_stream/IrUnitHandlerInterface.hpp
+          # - src/clp/ir/EncodedTextAst.cpp
+          # - tests/test-ffi_IrUnitHandlerInterface.cpp
```

```suggestion
          # NOTE:
          #
          # - We don't lint `.inc` files because they result in false positives due to circular
          #   includes, and we plan to remove them going forward.
          # - The only error in the following files is a circular include due to including
          #   decoding_methods.hpp (decoding_methods.hpp and decoding_methods.inc include each
          #   other); the files should be removed from the exclude list once decoding_methods.inc
          #   is removed:
          #   - src/clp/ffi/ir_stream/ir_unit_deserialization_methods.hpp
          #   - src/clp/ffi/ir_stream/IrUnitHandlerInterface.hpp
          #   - src/clp/ir/EncodedTextAst.cpp
          #   - tests/test-ffi_IrUnitHandlerInterface.cpp
```

Formatting might be excessive. Feel free to disagree.